### ISSUE TITLE
Handle incoming verification requests

### ISSUE THREAD

**Issue by tulir** [tulir/gomuks#205]

**Comment by languitar** [comment]
Is there a workaround to validate gomuks?

**Comment by tim-clifford** [comment]
there are (I imagine) several workarounds. This worked for me:
from gomuks: `/fingerprint`
from element android: *Settings -> Security & Privacy -> Active Sessions -> Show All Sessions -> gomuks -> Manually verify by text*. 
Check the fingerprints match and click verify.

**Comment by languitar** [comment]
I must be blind. I don't have a "Manually verify by text" in Element.

**Comment by languitar** [comment]
aaah, that only exists on android :see_no_evil:

**Comment by tulir** [comment]
It exists on web too, you just have to use the user list to view your sessions and then click on the specific session.

You can also use `/cs fetch` + `/cs self-sign` to use cross-signing to verify yourself without another device

**Comment by languitar** [comment]
> It exists on web too, you just have to use the user list to view your sessions and then click on the specific session. 

That's pretty well hidden and in an unexpected place. Verification of own session shouldn't be related to UI that only exists for specific rooms...

In any case: if implementation of interactive verification takes some more time, maybe a note in the documentation could help to avoid this confusion in the future?

**Comment by proycon** [comment]
Maybe something has changed in Element in the meantime, but I really can't find any of the ways to manually verify by text. What is the current proper workaround to verify the gomuks session?

**Comment by mcepl** [comment]
> Maybe something has changed in Element in the meantime, but I really can't find any of the ways to manually verify by text. What is the current proper workaround to verify the gomuks session?

This https://github.com/tulir/gomuks/issues/205#issuecomment-998927759 should work.

**Comment by proycon** [comment]
Not really, I tried that.. `/cs fetch` It prompts for a "passphrase", but neither my account password nor the "security key" (from Riot/Element) works. `/cs status` says:  Cross-signing keys are published, but private keys are not cached

I also exported E2E room keys from Element and then used `/import` in gomuks (which also prompted for a passphrase, but here my account password did work fine). 

But whatever I try the gomuks sessions remain 'unverified' in Element.

**Comment by quite** [comment]
Same issue as proycon. Would appreciate a complete workflow on how to get gomuks cross-signed and e2ee just working when using it as a secondary client in addition to Element.

**Comment by Evidlo** [comment]
What is the "passphrase" here?  When I set up my account, I saved a recovery key, but never set any passphrase.

![12_Setup-Key_en](https://github.com/tulir/gomuks/assets/5455841/61d8c4ea-5093-484b-9790-5f2903e19f6a)

**Comment by 3hg4hj** [comment]
> It exists on web too, you just have to use the user list to view your sessions and then click on the specific session.
> 
> You can also use `/cs fetch` + `/cs self-sign` to use cross-signing to verify yourself without another device

`Error fetching cross-signing keys: illegal base64 data at input byte 56`
